src/hooks/useExport.ts

```diff
@@ -0,0 +1,100 @@
+import html2canvas from 'html2canvas'
+import { useCallback, useState } from 'react'
+
+export type CopyState = 'idle' | 'preparing' | 'success' | 'failure'
+
+export type DownloadState = 'idle' | 'preparing' | 'failure'
```

Comment:
I guess it makes sense not having success state here
vs
Needing a success state on copy so there is a noticeable side-effect.

Reply:
yeah, exactly! there's a one second timer that changes the button text to "copied"